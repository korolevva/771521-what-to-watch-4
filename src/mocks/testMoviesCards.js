export const moviesCards = [
  {
    id: 1,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    imagePreview: `img/bohemian-rhapsody.jpg`,
    poster: `https://placeimg.com/270/410/arch/grayscale`,
    title: `Bohemian Rhapsody`,
    genre: `Biography`,
    date: 2018,
    description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury`,
    rating: 8.0,
    ratingCount: 305,
    director: `Bryan Singer`,
    stars: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    duration: 167,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 2,
    background: `https://placeimg.com/1300/552/tech`,
    imagePreview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster: `https://placeimg.com/270/410/arch/grayscale`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Drama`,
    date: 2018,
    description: `The second installment of the "Fantastic Beasts" series featuring the adventures of Magizoologist Newt Scamander.`,
    rating: 6.6,
    ratingCount: 423,
    director: `David Yates`,
    stars: [`Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`],
    duration: 90,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];
