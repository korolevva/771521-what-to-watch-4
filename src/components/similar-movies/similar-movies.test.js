import * as React from "react";
import renderer from "react-test-renderer";
import SimilarMovies from "./similar-movies.jsx";
import {moviesCards} from "../../mocks/testMoviesCards.js";
import {Router} from "react-router-dom";
import history from "../../history.js";

it(`Render SimilarMovies`, () => {
  const tree = renderer
    .create(
        <Router
          history={history}
        >
          <SimilarMovies
            moviesCards={moviesCards}
          />
        </Router>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
