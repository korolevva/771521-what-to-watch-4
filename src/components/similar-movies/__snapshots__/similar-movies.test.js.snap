// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SimilarMovies 1`] = `
<section
  className="catalog catalog--like-this"
>
  <h2
    className="catalog__title"
  >
    More like this
  </h2>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        href="/film/1"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            autoPlay={false}
            controls={false}
            height="175"
            muted={true}
            poster="img/bohemian-rhapsody.jpg"
            width="280"
          >
            <source
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              type="video/mp4"
            />
          </video>
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/film/1"
          onClick={[Function]}
        >
          Bohemian Rhapsody
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        href="/film/2"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            autoPlay={false}
            controls={false}
            height="175"
            muted={true}
            poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
            width="280"
          >
            <source
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              type="video/mp4"
            />
          </video>
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/film/2"
          onClick={[Function]}
        >
          Fantastic Beasts: The Crimes of Grindelwald
        </a>
      </h3>
    </article>
  </div>
</section>
`;
